lib/index.js

```diff
@@ -1,70 +1,58 @@
 'use strict';
 
-/**
- * Module dependencies.
- */
-
 var integration = require('@segment/analytics.js-integration');
+var useHttps = require('use-https');
 
 /**
- * Expose `Hindsight` integration.
+ * Expose `Hindsight`
+ *
  */
 
 var Hindsight = module.exports = integration('Hindsight')
-  .option('apiKey', '')
-  .tag('<script src="">');
+  .global('RB')
+  .option('pixel_code','')
+  .tag('http', '<script src="http://getrockerbox.com/assets/xyz.js">')
+  .tag('https', '<script src="https://getrockerbox.com/assets/xyz.js">');
 
 /**
- * Initialize.
- *
- * @api public
+ * Initialize
  */
 
 Hindsight.prototype.initialize = function() {
-  // put your initialization logic here
+  window.RB = {};
+  window.RB.disablePushState = true;
+  window.RB.queue = [];
+  window.RB.track = window.RB.track || function() {
+    window.RB.queue.push(Array.prototype.slice.call(arguments));
+  };
+  window.RB.source = this.options.pixel_code;
+
+  var protocol = useHttps() ? 'https' : 'http';
 
-  this.load(this.ready);
+  this.load(protocol, this.ready);
 };
 
 /**
- * Loaded?
- *
- * @api public
- * @return {boolean}
+ * Loaded
  */
 
 Hindsight.prototype.loaded = function() {
-  // what are required properties or functions that you need available on the `window`
-  // before the integration marks itself as ready?
-  return true;
+  return !!window.RB && !!window.RB.track;
```

Comment:
@patrickotoole it's better to assert against some object that is replaced/added only _after_ your script downloads. these are things that you already stub/create inside the initialize function, rendering this function a bit useless

a common one is to check that the `push` functions of your queue is `!==` Array's prototype push function